Reset should unassign Stat Points (previously known as Attribute Points) #12123

PitiTheGrey opened the pull request:
Fixes https://github.com/HabitRPG/habitica/issues/12123

PitiTheGrey committed:
Update reset.js

Stat points are set to zero on reset

PitiTheGrey committed:
Update reset.js

Test if habitica/website/common/script/ops/reset.js sets stat points to zero.

Alys commented on test/common/ops/reset.js:
```diff
@@ -52,6 +52,20 @@ describe('shared.ops.reset', () => {
+    user.stats.str = 2;
+    user.stats.con = 2;
+    user.stats.int = 2;
+    user.stats.per = 2;
```
Could you please add stats.points to this and to the expect() statements below? :)

In case you're not aware, you can make another commit on the same branch in your local install and push it to your github fork, and it will automatically appear in this PR.

PitiTheGrey replied:
Thank you for your support and encouragement, Alys! You are super!

PitiTheGrey committed:
Update reset.js

Tests if str, con, int, per are set to 0 and points are set to 1 on habitica/website/common/script/ops/reset.js

paglias commented:
Hi @PitiTheGrey , I'm not exactly sure what went wrong but your fork of Habitica seems to be a bit out of date:

<img width="634" alt="Schermata 2020-04-28 alle 12 51 47" src="https://user-images.githubusercontent.com/589911/80479259-0a82fe00-894f-11ea-95a7-56b49d9982b8.png">

And it seems to have resulted in changes to the reset op to be applied on an old version of the code, you can see in the diff for example:

<img width="1250" alt="Schermata 2020-04-28 alle 12 52 40" src="https://user-images.githubusercontent.com/589911/80479340-2a1a2680-894f-11ea-983e-e58540721b9b.png">

But the current code is already using `export default` https://github.com/HabitRPG/habitica/blob/develop/website/common/script/ops/reset.js#L4

Could you try to update your fork to the current version of the `develop` branch in Habitica and update the PR? Thanks

PitiTheGrey committed:
Merge pull request #2 from HabitRPG/develop

update

PitiTheGrey commented:
Please close this pull request.  
There is a new one with the correct database. ;)

Alys commented:
Closing in favour of https://github.com/HabitRPG/habitica/pull/12148
